### ISSUE TITLE
Stream game ID and name is nullable 

### ISSUE THREAD

**Issue by LostLuma** [twitchdev/issues#1033]
**Brief description**

The [get streams](https://dev.twitch.tv/docs/api/reference/#get-streams) response lists both `game_id` and `game` as just `String`, whereas they are nullable.

**Expected documentation**

The documentation should mention that these fields are nullable.

**Screenshots**

![image](https://github.com/user-attachments/assets/84342626-d9f7-47d4-a4e6-ae332b6d35f0)

**Additional context or questions**

Nullability / Optionality should just be an additional column in the table as it is in the request body section ... reading the docs is needlessly complicated.

**Comment by BarryCarlyon** [comment]
I cannot replicate this one, I get an empty string not a null

I imagine there could be an edge case for a _new account_ that never has set a category however?
So what call did you do to get this result?

Calling

`twitch api get 'streams?user_login=barrycarlyonbot'`

and fetch, same response:

```
{
    "data": [
        {
            "id": "43431849752",
            "user_id": "56410307",
            "user_login": "barrycarlyonbot",
            "user_name": "BarryCarlyonBot",
            "game_id": "",
            "game_name": "",
            "type": "live",
            "title": "Live for Testing of things and the stuff!",
```

**Comment by LostLuma** [comment]
You're right, it is an empty string instead of null which is even more annoying to work with and should definitely be documented.

**Comment by BarryCarlyon** [comment]
an empty string is _still_ a string. as `""` is type of `string`

This can go blank if

- the user has not set a game
- the game was deleted and the channel is still set to the removed game (need to check/test this one)



> should definitely be documented.

So you want the behaviour of what happens "if a channel has not set a game" documented?

so

> The ID of the category or game being played.

becomes

> The ID of the category or game being played, or empty string if a category/name is not set

similar to whats documented for title

> The stream’s title. Is an empty string if not set.

_conceptually_ it's super rare for someone to stream without setting title/category coz whats the point of streaming if you didn't set this :-P

**Comment by LostLuma** [comment]
I do want that documented, yes. It allows people to write correct code that won't run into any errors for edge-case streams.

**Comment by thedist** [comment]
Out of curiosity, what were you doing that would be erroring unless you were making assumptions about the content of the string? As you may run in to issues elsewhere in the docs if assumptions are made about what type of format strings will take.

**Comment by EvieePy** [comment]
There's no game called "" and no game with an id of "". No assumptions are being made, why would it be too difficult for the docs to simply say: "Could be an empty string if no game has been set". I honestly don't even know why this an argument/debate?

**Comment by LostLuma** [comment]
I'm not assuming anything. I read the docs and the docs did not say that this field may not be a valid category / game.  

If this is an issue elsewhere as well then this should be fixed everywhere because the docs are supposed to document how the API *actually works* and help you write better programs. If they fail to do so their usefulness is diminished by a huge factor.

**Comment by Yoyoyopo5** [comment]
I understand the concern to some extent. Without an explicit documentation, the behavior for getting a _game_id_ and _game_name_ on a stream that doesn't have a category selected is assumed undefined. While I believe the behavior is defined, it requires testing to discover.

If the behavior was entirely consistent across all endpoints, it could be listed as a [concept](https://dev.twitch.tv/docs/api/guide/) and would be a non-issue. But there are multiple instances of inconsistent empty vs null string usage. (e.g., [Get Ad Schedule](https://dev.twitch.tv/docs/api/reference/#get-ad-schedule) uses an empty string for _last_ad_at_ if an ad has not been played or channel isn't live, while [Get Custom Reward](https://dev.twitch.tv/docs/api/reference/#get-custom-reward) uses a null for _cooldown_expires_at_ if there is no cooldown set).

Although not including it wouldn't be the end of the world, it should be simple enough to add another sentence to this endpoint describing the behavior if a category is not set on a stream.

**Comment by Xemdo** [comment]
Documentation has been updated

**Comment by LostLuma** [comment]
Thank you!
